include/layers/FCLayer.hpp

```diff
@@ -2,27 +2,78 @@
 #include <stdexcept>
 #include <vector>
 
+template <typename T>
+class Shape2D {
```

Comment:
Shapes can be not only 2D. Can we consider that?

Reply:
Now nD Shape is possible